code/modules/mob/interactive.dm

```diff
@@ -342,31 +388,32 @@
 					tryWalk(T)
 			//THIEVING SKILLS
 			if(prob(slyness))
-				//---------TOOLS
-				if(istype(TARGET, /obj/item/weapon))
-					var/obj/item/weapon/W = TARGET
-					if(W.force >= best_force || prob((FUZZY_CHANCE_LOW+FUZZY_CHANCE_HIGH)/2))
-						if(!l_hand || !r_hand)
-							take_to_slot(W)
-						else
-							insert_into_backpack()
-				//---------FASHION
-				if(istype(TARGET,/obj/item/clothing))
-					if(prob((FUZZY_CHANCE_LOW+FUZZY_CHANCE_HIGH)/2))
-						if(!l_hand || !r_hand)
-							var/obj/item/clothing/C = TARGET
-							take_to_slot(C)
+				if(!TARGET in blacklistItems)
+					//---------TOOLS
+					if(istype(TARGET, /obj/item/weapon))
+						var/obj/item/weapon/W = TARGET
+						if(W.force >= best_force || prob((FUZZY_CHANCE_LOW+FUZZY_CHANCE_HIGH)/2))
+							if(!l_hand || !r_hand)
+								take_to_slot(W)
+							else
+								insert_into_backpack()
+					//---------FASHION
+					if(istype(TARGET,/obj/item/clothing))
+						var/obj/item/clothing/C = TARGET
+						drop_item()
+						spawn(5)

```

Comment:
Accounting for server lag/other various update times. Without a spawn it seemed to bug out a lot.